src/Core/Worker.php

```diff
@@ -445,42 +445,98 @@ public static function execute(array $queue): bool
-			sleep($cooldown);
+		if (($load_cooldown == 0) && ($processes_cooldown == 0)) {
+			return false;
 		}
 
+		$load = System::getLoadAvg();
+		if (empty($load)) {
+			return false;
+		}
+
+		if (($load_cooldown > 0) && ($load['average1'] > $load_cooldown)) {
+			return true;
+		}
+
+		if (($processes_cooldown > 0) && ($load['scheduled'] > $processes_cooldown)) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Slow the execution down if the system load is too high
+	 *
+	 * @return void
+	 */
+	public static function coolDown()
+	{
 		$load_cooldown      = DI::config()->get('system', 'worker_load_cooldown');
 		$processes_cooldown = DI::config()->get('system', 'worker_processes_cooldown');
 
-		while ((($load_cooldown > 0) || ($processes_cooldown > 0)) && ($load = System::getLoadAvg())) {
+		if (($load_cooldown == 0) && ($processes_cooldown == 0)) {
+			return;
+		}
+
+		$sleeping = false;
+
+		while ($load = System::getLoadAvg()) {
 			if (($load_cooldown > 0) && ($load['average1'] > $load_cooldown)) {
-				Logger::debug('Load induced pre execution cooldown.', ['max' => $load_cooldown, 'load' => $load, 'id' => $queue['id'], 'priority' => $queue['priority'], 'command' => $queue['command']]);
+				if (!$sleeping) {
+					Logger::notice('Load induced pre execution cooldown.', ['max' => $load_cooldown, 'load' => $load, 'called-by' => System::callstack(1)]);
+					$sleeping = true;
+				}
 				sleep(1);
 				continue;
 			}
 			if (($processes_cooldown > 0) && ($load['scheduled'] > $processes_cooldown)) {
-				Logger::debug('Process induced pre execution cooldown.', ['max' => $processes_cooldown, 'load' => $load, 'id' => $queue['id'], 'priority' => $queue['priority'], 'command' => $queue['command']]);
+				if (!$sleeping) {
+					Logger::notice('Process induced pre execution cooldown.', ['max' => $processes_cooldown, 'load' => $load, 'called-by' => System::callstack(1)]);
+					$sleeping = true;
+				}
 				sleep(1);
 				continue;
 			}
 			break;
 		}
 
+		if ($sleeping) {
+			Logger::notice('Cooldown ended.', ['max-load' => $load_cooldown, 'max-processes' => $processes_cooldown, 'load' => $load, 'called-by' => System::callstack(1)]);
+		}
+	}
+
+	/**
+	 * Execute a function from the queue
+	 *
+	 * @param array   $queue       Workerqueue entry
+	 * @param string  $funcname    name of the function
+	 * @param array   $argv        Array of values to be passed to the function
+	 * @param boolean $method_call boolean
+	 * @return void
+	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 */
+	private static function execFunction(array $queue, string $funcname, array $argv, bool $method_call)
+	{
+		$a = DI::app();
+
+		$cooldown = DI::config()->get('system', 'worker_cooldown', 0);
+		if ($cooldown > 0) {
+			Logger::notice('Pre execution cooldown.', ['cooldown' => $cooldown, 'id' => $queue['id'], 'priority' => $queue['priority'], 'command' => $queue['command']]);
+			sleep($cooldown);
+		}
```

Comment:
Shouldn't this be included in `coolDown()`?

Reply:
I thought about this as well. But I decided against this. Now I'm currently thinking about, if that could be an idea ...

Reply:
Changed